EventPhases/explainer.md

```diff
@@ -0,0 +1,501 @@
+
+The difficulty in lining up these exclusions as to be useful appears difficult, but we welcome feedback on this.
+
+### Parallel, Async APIs
+
+Another way around the problems of synchronous layouts and computed style information could be the introduction of parallel, fully asynchronous APIs with options for disabling synchronous variants. For example, a promise-vending API for delivering element geometry information (similar to [Resize Observers](https://web.dev/articles/resize-observer)) would give runtimes the ability to almost-synchronously deliver that information inside the same turn when the DOM is not dirtied (modified in a way that would affect layout), or defer delivery to the beginning of the very next turn after a layout has completed. There are questions to be litigated about staleness of layout data relative to promise resolution, but a simple variant of such an API might integrate with the high-performance [CSS Typed OM](https://developer.chrome.com/docs/css-ui/cssom) like this:
+
+```html
+<!-- 
+ Opt-in globally disables old APIs, exposes new ones 
+-->
+<meta http-equiv="Document-Policy" 
+      content="synchronous-style-accessors: disable">
+
+<script type="module">
+  let foo = document.querySelector("#foo");
+
+  console.log(foo.innerWidth === undefined); // true
+
+  // A new shorthand for getting the computedStyleMap; 
+  // ordinarily calling `window.getComputedStyles(foo)` 
+  // or `foo.getComputedStyleMap()` would force
+  // synchronous re-resolution of all dirtied styles. 
+  // This variant returns a promise that returns in 
+  // the next microtask (assuming no DOM modification 
+  // has happened since turn start) or at 
+  // the beginning of the next turn:
+  let computed = await foo.computedStyles();
+
+  // A new shorthand for accessing geometry.
+  // This object provides access to all geometry, 
+  // including (but not limited to):
+  //  - widths (margin, border, content, etc.)
+  //  - heights
+  //  - scroll dimensions
+  //  - offsets
+  // This alternative to calling accessors such as 
+  // `offsetWidth` allows the engine to re-use existing 
+  // geometry computed from the last layout, or defer 
+  // the lines following until the next turn (after a 
+  // layout has occurred naturally):
+  let dimensions = await foo.dimensions();
+  let bw = dimensoins.borderWidth;
+
+  foo.attributeStyleMap.set("width", CSS.px(bw + 10));
+</script>
+```
+
+This style of API opt-in is unusual on the web platform, and our experience designing [the updated DOM for Dart (`dart:html`)](https://dart.dev/libraries/dart-html) has convinced us that the required of transition is exceedingly difficult to pull off in practice, [no matter how nice the new APIs are by comparison.](https://api.dart.dev/stable/latest/dart-html/Element/on.html)
```

Comment:
"that the required of transition" -> not sure exactly what this is meant to say, but maybe "that the work required of transitioning"?